File: crates/iota-sdk/examples/get_transaction.rs
// Copyright (c) 2025 IOTA Stiftung
// SPDX-License-Identifier: Apache-2.0

use iota_sdk::graphql_client::{Client, error::GraphQLResult, pagination::PaginationFilter};

#[tokio::main]
async fn main() -> GraphQLResult<()> {
    let client = Client::new_localnet();

    let transactions = client
        .transactions(None, PaginationFilter::default())
        .await?;
    let digest = transactions
        .data()
        .first()
        .expect("no transactions found")
        .transaction
        .digest();

    let signed_transaction = client.transaction(digest).await?.expect("tx not found");
    println!("Signed Transaction: {signed_transaction:#?}\n");

    let transaction_effects = client
        .transaction_effects(digest)
        .await?
        .expect("tx not found");
    println!("Transaction Effects: {transaction_effects:#?}\n");

    let transaction_data_effects = client
        .transaction_data_effects(digest)
        .await?
        .expect("tx not found");
    println!("Transaction Data Effects: {transaction_data_effects:#?}");

    Ok(())
}

File: bindings/wasm/examples/get_transaction.mjs
// Copyright (c) 2026 IOTA Stiftung
// SPDX-License-Identifier: Apache-2.0

import { GraphQlClient, initAsync } from "@iota/sdk-wasm";

await initAsync();

const client = GraphQlClient.newLocalnet();

const transactions = await client.transactions();
if (transactions.data.length === 0) {
  throw new Error("No transactions found");
}
const digest = transactions.data[0].transaction.digest();

const signedTransaction = await client.transaction(digest);
console.log(`Signed Transaction: \`${signedTransaction}\`\n`);

const transactionEffects = await client.transactionEffects(digest);
console.log(`Transaction Effects: \`${transactionEffects}\`\n`);

const transactionDataEffects = await client.transactionDataEffects(digest);
console.log(`Transaction Data Effects: \`${transactionDataEffects}\`\n`);
